src/Widgets/DwellClickTimer.vala

```diff
@@ -0,0 +1,139 @@
+
+            transition = new Clutter.PropertyTransition ("angle");
+            transition.set_progress_mode (Clutter.AnimationMode.EASE_OUT_QUAD);
+            transition.set_animatable (this);
+            transition.set_from_value (START_ANGLE);
+            transition.set_to_value (START_ANGLE + (2 * Math.PI));
+
+            transition.new_frame.connect (() => {
+                queue_redraw ();
+            });
+
+            var rgba = Gdk.RGBA ();
+            rgba.parse (BACKGROUND_COLOR);
+            stroke_color = new Cairo.Pattern.rgb (rgba.red, rgba.green, rgba.blue);
+            fill_color = new Cairo.Pattern.rgba (rgba.red, rgba.green, rgba.blue, BACKGROUND_OPACITY);
+
+            scaling_factor = InternalUtils.get_ui_scaling_factor ();
+            set_size (WIDTH_PX * scaling_factor, HEIGHT_PX * scaling_factor);
+
+            var seat = Clutter.get_default_backend ().get_default_seat ();
+            seat.set_pointer_a11y_dwell_click_type (Clutter.PointerA11yDwellClickType.PRIMARY);
+
+            seat.ptr_a11y_timeout_started.connect ((device, type, timeout) => {
+                var tracker = wm.get_display ().get_cursor_tracker ();
+                int x, y;
+                tracker.get_pointer (out x, out y, null);
+
+                this.x = x - (width / 2);
+                this.y = y - (width / 2);
+
+                transition.set_duration (timeout);
+                visible = true;
+                transition.start ();
+            });
+
+            seat.ptr_a11y_timeout_stopped.connect ((device, type, clicked) => {
+                transition.stop ();
+                visible = false;
+            });
+        }
+
+        public override void paint (Clutter.PaintContext context) {
+            var width = WIDTH_PX * scaling_factor;
+            var height = HEIGHT_PX * scaling_factor;
+
+            var radius = int.min (width / 2, height / 2);
+            var end_angle = START_ANGLE + angle;
+
+            var surface = new Cairo.ImageSurface (Cairo.Format.ARGB32, width, height);
+            var cr = new Cairo.Context (surface);
+            cr.set_line_cap (Cairo.LineCap.ROUND);
+            cr.set_line_join (Cairo.LineJoin.ROUND);
+            cr.translate (width / 2, height / 2);
+
+            cr.move_to (0, 0);
+            cr.arc (0, 0, radius - BORDER_WIDTH_PX * scaling_factor, START_ANGLE, end_angle);
+            cr.line_to (0, 0);
+            cr.close_path ();
+
+            cr.set_line_width (0);
+            cr.set_source (fill_color);
+            cr.fill_preserve ();
+
+            cr.set_line_width (BORDER_WIDTH_PX * scaling_factor);
+            cr.set_source (stroke_color);
+            cr.stroke ();
+
+            var cogl_context = context.get_framebuffer ().get_context ();
+
+            try {
+                var texture = new Cogl.Texture2D.from_data (cogl_context, width, height, Cogl.PixelFormat.BGRA_8888_PRE,
```

Comment:
Creating a new texture like this should really only happen when the width or height changes, otherwise there's no reason to recreate it everytime on paint.

Reply:
Thanks, yeah, that makes sense. Done!